tools/swift-inspect/Sources/swift-inspect/WindowsRemoteProcess.swift

```diff
@@ -233,37 +237,286 @@ internal final class WindowsRemoteProcess: RemoteProcess {
+    let waitTimeoutMs = DWORD(WAIT_TIMEOUT_MS)
+
+    // Set up the shared memory
+    let hMapFile = CreateFileMappingA(INVALID_HANDLE_VALUE,
+                                      UnsafeMutablePointer(bitPattern: 0),
+                                      DWORD(PAGE_READWRITE),
+                                      0,
+                                      DWORD(bufSize),
+                                      sharedMemoryName);
+    if hMapFile == HANDLE(bitPattern: 0) {
+      print("CreateFileMapping failed \(GetLastError())")
+      return
+    }
+    if GetLastError() == ERROR_ALREADY_EXISTS {
+      print("FileMapping in use. Try again in a while.")
+      CloseHandle(hMapFile)
+      return
+    }
+    defer { CloseHandle(hMapFile) }
+    let FILE_MAP_ALL_ACCESS = STANDARD_RIGHTS_REQUIRED | SECTION_QUERY |
+                              SECTION_MAP_WRITE | SECTION_MAP_READ |
+                              SECTION_MAP_EXECUTE|SECTION_EXTEND_SIZE
+    let buf: LPVOID = MapViewOfFile(hMapFile,
+                                    DWORD(FILE_MAP_ALL_ACCESS),
+                                    0,
+                                    0,
+                                    SIZE_T(bufSize));
+    if buf == LPVOID(bitPattern: 0) {
+      print("MapViewOfFile failed \(GetLastError())")
+      return
+    }
+    defer { UnmapViewOfFile(buf) }
+
+    // Set up the event objects
+    let hReadEvent: HANDLE = CreateEventA(LPSECURITY_ATTRIBUTES(bitPattern: 0),
+                                          false, // Auto-reset
+                                          false, // Initial state is nonsignaled
+                                          readEventName)
+    if hReadEvent == HANDLE(bitPattern: 0) {
+      print("CreateEvent failed \(GetLastError())")
+      return
+    }
+    defer { CloseHandle(hReadEvent) }
+    let hWriteEvent: HANDLE = CreateEventA(LPSECURITY_ATTRIBUTES(bitPattern: 0),
+                                           false, // Auto-reset
+                                           false, // Initial state is nonsignaled
+                                           writeEventName)
+    if hWriteEvent == HANDLE(bitPattern: 0) {
+      print("CreateEvent failed \(GetLastError())")
       return
     }
+    defer { CloseHandle(hWriteEvent) }
 
-    repeat {
-      var entry: HEAPENTRY32 = HEAPENTRY32()
-      entry.dwSize = SIZE_T(MemoryLayout<HEAPENTRY32>.size)
+    // Allocate the dll path string in the remote process.
+    guard let dllPathRemote = allocateDllPathRemote() else {
+      return
+    }
 
-      if !Heap32First(&entry, dwProcessId, heap.th32HeapID) {
-        // FIXME(compnerd) log error
-        continue
+    var hKernel32 = HMODULE(bitPattern: 0)
+    "kernel32".withCString() {
+      hKernel32 = GetModuleHandleA($0)
+    }
+
+    // Load the dll and start the heap walk
+    let hookAddr = unsafeBitCast(GetProcAddress(hKernel32, "LoadLibraryW"),
+                                 to: LPTHREAD_START_ROUTINE.self)
```

Comment:
This seems unnecessary.  It resolves to the local function `LoadLibraryW`, so why do the dynamic lookup?

Reply:
Casting the local function pointer `LoadLibraryW` to  `UnsafeMutableRawPointer` or `LPTHREAD_START_ROUTINE` doesn't seem to work with `Swift/Builtin.swift:95: Fatal error: Can't unsafeBitCast between types of different sizes`. How?

Reply:
```swift
let lpfnLoadLibraryW: @convention(c) (LPCWSTR?) -> HMODULE? = LoadLibraryW
_ = unsafeBitCast(lpfnLoadLibraryW, to: LPTHREAD_START_ROUTINE.self)
```

I imagine you can even combine the two lines.